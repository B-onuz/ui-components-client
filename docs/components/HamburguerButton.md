# HamburguerButton (Component)

Botão de hamburguer, é utilizado no layout base.

## Como utilizar

```js
import HamburguerButton from 'ui-components/lib/components/HamburguerButton'
```

```jsx
import React, { useState } from 'react'
import HamburguerButton from 'ui-components/lib/components/HamburguerButton'

const HamburguerButtonExample = () => {
  const [isOpen, setIsOpen] = useState(false)
  return <HamburguerButton onClick={() => setIsOpen(!isOpen)} isOpen={isOpen} />
}
```

## Styled-system props

- space https://styled-system.com/table#space
- layout https://styled-system.com/table#layout
