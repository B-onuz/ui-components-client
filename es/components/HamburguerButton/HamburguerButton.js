"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/HamburguerButton/HamburguerButton.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 0;\n  bottom: 0;\n  margin: auto;\n  right: -5px;\n  transition: 0.15s linear;\n  transition-delay: 0.35s;\n  transform: rotate(135deg) scale(0);\n  opacity: 0.3;\n  transform-origin: center center;\n  z-index: 10;\n  width: 15px;\n  height: 15px;\n  ", "\n  &::after, &::before {\n    content: '';\n    border: solid;\n    border-width: 0 5px 5px 0;\n    display: inline-block;\n    padding: 5px;\n    transform: rotate(135deg);\n    -webkit-transform: rotate(135deg);\n    position: absolute;\n    top: 0;\n    right: 0;\n    bottom: 0;\n    left: 0;\n    margin: auto;\n  }\n  &::before {\n    z-index: 1;\n    border-width: 8px 8px 8px 8px;\n    border-color: ", ";\n    background: ", ";\n    margin-top: -5px;\n    margin-left: -2px;\n    border-radius: 4px;\n    margin-right: -8px;\n  }\n  &::after {\n    z-index: 2;\n    border-color: #333333de;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  width: 35px;\n  height: 26px;\n  border: none;\n  position: relative;\n  background: linear-gradient(to bottom, #333333de, #333333de);\n  background-size: 100% 20%;\n  background-repeat: no-repeat;\n  background-position: center center;\n  transition: background-size 0.2s 0.25s ease;\n  outline: none;\n  cursor: pointer;\n  transform: scale(0.6);\n  transition: background-size 0.3s 0.2s ease;\n  &:before,\n  &:after {\n    height: 20%;\n    width: 100%;\n    position: absolute;\n    left: 0;\n    background-color: #333333de;\n    content: '';\n    transition: transform 0.3s ease;\n  }\n  &:before {\n    top: 0;\n  }\n  &:after {\n    bottom: 0;\n  }\n  &:before,\n  &:after {\n    transition: 0.3s ease;\n  }\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledHamburguerButton = _styledComponents["default"].button(_templateObject(), function (props) {
  return props.isOpen && "\n      // background-size: 0 0;\n      &:before,\n      &:after {\n        transition-delay: 0.3s;\n      }\n      // &:before {\n      //   transform: translateY(200%) rotate(45deg);\n      // }\n      // &:after {\n      //   transform: translateY(-200%) rotate(-45deg);\n      // }\n    ";
});

var Arrow = _styledComponents["default"].i(_templateObject2(), function (props) {
  return props.isOpen && " opacity: 1;\n      transform: rotate(0) scale(1);\n    ";
}, function (props) {
  return props.backgroundColor;
}, function (props) {
  return props.backgroundColor;
});

var HamburguerButton = function HamburguerButton(_ref) {
  var _ref$isOpen = _ref.isOpen,
      isOpen = _ref$isOpen === void 0 ? false : _ref$isOpen,
      children = _ref.children,
      _ref$backgroundColor = _ref.backgroundColor,
      backgroundColor = _ref$backgroundColor === void 0 ? '#fff' : _ref$backgroundColor,
      rest = _objectWithoutPropertiesLoose(_ref, ["isOpen", "children", "backgroundColor"]);

  return /*#__PURE__*/_react["default"].createElement(StyledHamburguerButton, _extends({
    isOpen: isOpen
  }, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 108,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react["default"].createElement(Arrow, {
    className: "arrow left",
    isOpen: isOpen,
    backgroundColor: backgroundColor,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 109,
      columnNumber: 7
    }
  }));
};

var _default = HamburguerButton;
exports["default"] = _default;