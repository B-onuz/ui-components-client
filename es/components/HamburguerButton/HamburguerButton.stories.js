'use strict'

function _typeof(obj) {
  '@babel/helpers - typeof'
  if (typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol') {
    _typeof = function _typeof(obj) {
      return typeof obj
    }
  } else {
    _typeof = function _typeof(obj) {
      return obj && typeof Symbol === 'function' && obj.constructor === Symbol && obj !== Symbol.prototype ? 'symbol' : typeof obj
    }
  }
  return _typeof(obj)
}

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.HamburguerButtonExample = exports['default'] = void 0

var _react = _interopRequireWildcard(require('react'))

var _HamburguerButton = _interopRequireDefault(require('./HamburguerButton'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/HamburguerButton/HamburguerButton.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

function _getRequireWildcardCache() {
  if (typeof WeakMap !== 'function') return null
  var cache = new WeakMap()
  _getRequireWildcardCache = function _getRequireWildcardCache() {
    return cache
  }
  return cache
}

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj
  }
  if (obj === null || (_typeof(obj) !== 'object' && typeof obj !== 'function')) {
    return { default: obj }
  }
  var cache = _getRequireWildcardCache()
  if (cache && cache.has(obj)) {
    return cache.get(obj)
  }
  var newObj = {}
  var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor
  for (var key in obj) {
    if (Object.prototype.hasOwnProperty.call(obj, key)) {
      var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null
      if (desc && (desc.get || desc.set)) {
        Object.defineProperty(newObj, key, desc)
      } else {
        newObj[key] = obj[key]
      }
    }
  }
  newObj['default'] = obj
  if (cache) {
    cache.set(obj, newObj)
  }
  return newObj
}

var _default = {
  title: 'HamburguerButton',
  component: _HamburguerButton['default'],
}
exports['default'] = _default

var HamburguerButtonExample = function HamburguerButtonExample() {
  var _useState = (0, _react.useState)(false),
    isOpen = _useState[0],
    setIsOpen = _useState[1]

  return /*#__PURE__*/ _react['default'].createElement(_HamburguerButton['default'], {
    onClick: function onClick() {
      return setIsOpen(!isOpen)
    },
    isOpen: isOpen,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 11,
      columnNumber: 10,
    },
  })
}

exports.HamburguerButtonExample = HamburguerButtonExample
