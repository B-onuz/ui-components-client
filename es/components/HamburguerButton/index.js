"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _HamburguerButton = _interopRequireDefault(require("./HamburguerButton"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _HamburguerButton["default"];
exports["default"] = _default;