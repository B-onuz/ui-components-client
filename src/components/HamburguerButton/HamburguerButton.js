import React from 'react'
import styled from 'styled-components'
import { color } from 'styled-system'

const StyledHamburguerButton = styled.button`
  width: 35px;
  height: 26px;
  border: none;
  position: relative;
  background: linear-gradient(to bottom, #F14F5C, #F14F5C);
  background-size: 100% 20%;
  background-repeat: no-repeat;
  background-position: center center;
  transition: background-size 0.2s 0.25s ease;
  outline: none;
  cursor: pointer;
  transform: scale(0.6);
  transition: background-size 0.3s 0.2s ease;
  &:before,
  &:after {
    height: 20%;
    width: 100%;
    position: absolute;
    left: 0;
    background-color: #F14F5C;
    content: '';
    transition: transform 0.3s ease;
  }
  &:before {
    top: 0;
  }
  &:after {
    bottom: 0;
  }
  &:before,
  &:after {
    transition: 0.3s ease;
  }
  ${(props) =>
    props.isOpen &&
    `
      // background-size: 0 0;
      &:before,
      &:after {
        transition-delay: 0.3s;
      }
      // &:before {
      //   transform: translateY(200%) rotate(45deg);
      // }
      // &:after {
      //   transform: translateY(-200%) rotate(-45deg);
      // }
    `}
`

const Arrow = styled.i`
  position: absolute;
  top: 0;
  bottom: 0;
  margin: auto;
  right: -5px;
  transition: 0.15s linear;
  transition-delay: 0.35s;
  transform: rotate(135deg) scale(0);
  opacity: 0.3;
  transform-origin: center center;
  z-index: 10;
  width: 15px;
  height: 15px;
  ${(props) =>
    props.isOpen &&
    ` opacity: 1;
      transform: rotate(0) scale(1);
    `}
  &::after, &::before {
    content: '';
    border: solid;
    border-width: 0 5px 5px 0;
    display: inline-block;
    padding: 5px;
    transform: rotate(135deg);
    -webkit-transform: rotate(135deg);
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    margin: auto;
  }
  &::before {
    z-index: 1;
    border-width: 8px 8px 8px 8px;
    border-color: ${(props) => props.backgroundColor};
    background: ${(props) => props.backgroundColor};
    margin-top: -5px;
    margin-left: -2px;
    border-radius: 4px;
    margin-right: -8px;
  }
  &::after {
    z-index: 2;
    border-color: #F14F5C;
  }
`

const HamburguerButton = ({ isOpen = false, children, backgroundColor = '#fff', ...rest }) => {
  return (
    <StyledHamburguerButton isOpen={isOpen} {...rest}>
      <Arrow className="arrow left" isOpen={isOpen} backgroundColor={backgroundColor} />
    </StyledHamburguerButton>
  )
}

export default HamburguerButton
