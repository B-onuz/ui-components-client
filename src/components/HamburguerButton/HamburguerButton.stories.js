import React, { useState } from 'react'
import HamburguerButton from './HamburguerButton'

export default {
  title: 'HamburguerButton',
  component: HamburguerButton,
}

export const HamburguerButtonExample = () => {
  const [isOpen, setIsOpen] = useState(false)
  return <HamburguerButton onClick={() => setIsOpen(!isOpen)} isOpen={isOpen} />
}
