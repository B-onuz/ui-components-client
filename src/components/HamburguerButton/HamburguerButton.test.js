import React from 'react'
import HamburguerButton from './HamburguerButton.js'
import renderer from 'react-test-renderer'

test('HamburguerButton changes the class when hovered', () => {
  const component = renderer.create(<HamburguerButton />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
