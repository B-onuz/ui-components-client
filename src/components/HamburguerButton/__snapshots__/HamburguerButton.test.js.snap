// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburguerButton changes the class when hovered 1`] = `
<button
  className="sc-AxjAm iYohGp"
>
  <i
    className="sc-AxirZ latOsZ arrow left"
  />
</button>
`;
